src/azure-cli/azure/cli/command_modules/backup/custom_help.py

```diff
@@ -0,0 +1,507 @@
+
+import time
+import json
+import re
+import os
+import time
+from datetime import datetime, timedelta
+from six.moves.urllib.parse import urlparse  # pylint: disable=import-error
+
+from knack.log import get_logger
+
+from msrest.paging import Paged
+from msrestazure.tools import parse_resource_id, is_valid_resource_id
+
+from azure.mgmt.recoveryservicesbackup.models import ProtectedItemResource, AzureIaaSComputeVMProtectedItem, \
+    AzureIaaSClassicComputeVMProtectedItem, ProtectionState, IaasVMBackupRequest, BackupRequestResource, \
+    BackupManagementType, WorkloadType, OperationStatusValues, JobStatus
+
+from azure.cli.core.util import CLIError, sdk_no_wait
+from azure.cli.command_modules.backup._client_factory import (
+    backup_protected_items_cf, job_details_cf, protection_container_refresh_operation_results_cf,
+    protection_containers_cf, backup_protectable_items_cf, resources_cf, backup_operation_statuses_cf,
+    protection_container_operation_results_cf)
+
+
+logger = get_logger(__name__)
+
+fabric_name = "Azure"
+os_windows = 'Windows'
+os_linux = 'Linux'
+password_offset = 33
+password_length = 15
+
+# Client Utilities
+
+
+def _is_native_name(name):
+    return ";" in name
+
+
+def _is_id(id):
+    return "/" in id
+
+
+def _is_sql(type):
+    return type.lower() == 'sqldatabase'
+
+
+def _is_hana(type):
+    return type.lower() == 'saphanadatabase'
+
+
+def _is_wl_container(name):
+    return 'vmappcontainer' in name.lower()
+
+
+def _is_range_valid(start_date, end_date):
+    if start_date > end_date:
+        raise CLIError(
+                    """
+                    Start date must be earlier than end date.
+                    """)
+
+
+def _get_resource_id(resource_id):
+    return "/".join(resource_id.split('/')[3:])
+
+
+def _get_target_path(type, path, logical_name, data_directory_paths):
+    for i in data_directory_paths:
+        if i.type == type:
+            data_directory_path = i
+    file_type = path.split('\\')[-1].split('.')[1]
+    file_name = logical_name + '_' + str(int(time.time())) + '.' + file_type
+    return data_directory_path.path + file_name
+
+
+def _get_containers(client, container_type, status, resource_group_name, vault_name, container_name=None):
+    filter_dict = {
+        'backupManagementType': container_type,
+        'status': status
+    }
+
+    if container_name and not _is_native_name(container_name):
+        filter_dict['friendlyName'] = container_name
+    filter_string = _get_filter_string(filter_dict)
+
+    paged_containers = client.list(vault_name, resource_group_name, filter_string)
+    containers = _get_list_from_paged_response(paged_containers)
+
+    if container_name and _is_native_name(container_name):
+        return [container for container in containers if container.name == container_name]
+
+    return containers
+
+
+def _get_protectable_item_for_vm(cli_ctx, vault_name, vault_rg, vm_name, vm_rg):
```

Comment:
remove the vm specific functions from this file

Reply:
Done